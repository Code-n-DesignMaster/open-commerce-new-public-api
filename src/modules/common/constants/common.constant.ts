export const VIEWER_FACTORY_TOKEN = 'VIEWER_FACTORY_TOKEN';
export const ROLLBAR_TOKEN = 'ROLLBAR_TOKEN';
