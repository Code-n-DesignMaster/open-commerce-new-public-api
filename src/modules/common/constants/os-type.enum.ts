export enum OS_TYPE {
  ANDROID = 'ANDROID',
  IOS = 'IOS',
}
