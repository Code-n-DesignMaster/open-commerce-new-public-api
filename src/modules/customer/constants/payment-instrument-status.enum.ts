export enum PAYMENT_INSTRUMENT_STATUS {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  PENDING = 'PENDING',
}
