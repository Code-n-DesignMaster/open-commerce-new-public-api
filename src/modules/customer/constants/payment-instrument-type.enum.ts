export enum PAYMENT_INSTRUMENT_TYPE {
  AMAZON_PAY = 'AMAZON_PAY',
  AMERICAN_EXPRESS = 'AMERICAN_EXPRESS',
  APPLE_PAY = 'APPLE_PAY',
  APPLE_PAY_VISA = 'APPLE_PAY_VISA',
  APPLE_PAY_MASTERCARD = 'APPLE_PAY_MASTERCARD',
  APPLE_PAY_AMERICAN_EXPRESS = 'APPLE_PAY_AMERICAN_EXPRESS',
  APPLE_PAY_DISCOVER = 'APPLE_PAY_DISCOVER',
  CARTES_BANCAIRES = 'CARTES_BANCAIRES',
  CARTE_BLANCHE = 'CARTE_BLANCHE',
  CHASE_PAY = 'CHASE_PAY',
  CHINA_UNION_PAY = 'CHINA_UNION_PAY',
  CITI_PAY = 'CITI_PAY',
  CREDIT = 'CREDIT',
  DEBIT = 'DEBIT',
  DINERS_CLUB = 'DINERS_CLUB',
  DISCOVER = 'DISCOVER',
  EFTPOS = 'EFTPOS',
  ELO = 'ELO',
  GIFT = 'GIFT',
  GOOGLE_WALLET = 'GOOGLE_WALLET',
  GOOGLE_PAY = 'GOOGLE_PAY',
  GOOGLE_PAY_VISA = 'GOOGLE_PAY_VISA',
  GOOGLE_PAY_MASTERCARD = 'GOOGLE_PAY_MASTERCARD',
  GOOGLE_PAY_AMERICAN_EXPRESS = 'GOOGLE_PAY_AMERICAN_EXPRESS',
  GOOGLE_PAY_DISCOVER = 'GOOGLE_PAY_DISCOVER',
  ID_CREDIT = 'ID_CREDIT',
  INTERAC = 'INTERAC',
  JCB = 'JCB',
  LASER = 'LASER',
  MAESTRO = 'MAESTRO',
  MASTERCARD = 'MASTERCARD',
  MASTERPASS = 'MASTERPASS',
  PAYPAL = 'PAYPAL',
  PREPAID = 'PREPAID',
  PRIVATE_LABEL = 'PRIVATE_LABEL',
  QUIC_PAY = 'QUIC_PAY',
  SOLO = 'SOLO',
  SUICA = 'SUICA',
  SWITCH = 'SWITCH',
  UNKNOWN = 'UNKNOWN',
  VENMO = 'VENMO',
  VISA = 'VISA',
  VISA_CHECKOUT = 'VISA_CHECKOUT',
  VISA_VPAY = 'VISA_VPAY',
}
