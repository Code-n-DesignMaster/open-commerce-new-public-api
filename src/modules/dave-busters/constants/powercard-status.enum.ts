export enum POWERCARD_STATUS_TYPE {
  DISABLED = 'DISABLED',
  INACTIVE = 'INACTIVE',
  OPEN = 'OPEN',
  STOLEN = 'STOLEN',
  VIP = 'VIP',
  VOID = 'VOID',
}
