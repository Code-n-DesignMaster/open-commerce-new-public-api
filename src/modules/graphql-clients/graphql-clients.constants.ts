export const CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const TRANSACTION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'TRANSACTION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN';

export const CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const TRANSACTION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'TRANSACTION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN';

export const GRAPHQL_SERVICE_DIRECTORY_TOKEN =
  'GRAPHQL_SERVICE_DIRECTORY_TOKEN';
