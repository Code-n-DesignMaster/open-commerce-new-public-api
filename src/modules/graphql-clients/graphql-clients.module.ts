import { Module, Global } from '@nestjs/common';
import { graphqlClientsProviders } from './graphql-clients.providers';

@Global()
@Module({
  providers: graphqlClientsProviders,
  exports: graphqlClientsProviders,
})
export class GraphqlClientsModule {}
