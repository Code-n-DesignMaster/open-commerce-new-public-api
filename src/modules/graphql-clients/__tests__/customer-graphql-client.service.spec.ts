import { CustomerGraphqlClientService } from '../clients/customer-graphql-client.service';
import { Test } from '@nestjs/testing';
import { CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN } from '../graphql-clients.constants';
import { LoggerService } from '@open-commerce/nestjs-logger';
import { mockGraphqlClientExecuteQuery } from './mock-graphql-client-execute-query';
import { mockCustomerResolveInfo } from './__mocks__/mock-customer-resolve-info';
import { CustomerPaymentInstrumentCreateDto } from '../../customer/dtos/customer-payment-instrument-create.dto';
import { mockPaymentInstrumentResolveInfo } from './__mocks__/mock-payment-instrument-resolve-info';
import { CustomerPaymentInstrumentUpdateDto } from '@open-commerce/data-objects';
import { ICustomerInternal } from '../../common/interfaces/customer-internal.interface';
import { CustomerRegistrationCompleteDto } from '../../auth/dtos/customer-registration-complete.dto';

const mockLogger = {
  debug: jest.fn(),
  warn: jest.fn(),
  info: jest.fn(),
  error: jest.fn(),
  log: jest.fn(),
};

const mockGraphqlClient = {
  executeQuery: jest.fn(),
};

const testCustomerUuid = 'testCustomerUuid';

describe('CustomerGraphqlClientService', () => {
  let service: CustomerGraphqlClientService;

  beforeAll(async () => {
    const module = await Test.createTestingModule({
      providers: [
        CustomerGraphqlClientService,
        {
          provide: LoggerService,
          useValue: mockLogger,
        },
        {
          provide: CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN,
          useValue: mockGraphqlClient,
        },
      ],
    }).compile();

    service = module.get(CustomerGraphqlClientService);
  });

  describe('clientPaymentToken', () => {
    const customerId = 'testCustomerId';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('clientPaymentToken', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.clientPaymentToken(customerId);
    });
  });

  describe('customerByIdentityProviderId', () => {
    const identityProviderId = 'testIdentityProviderId';

    beforeEach(() => {
      mockGraphqlClientExecuteQuery(
        'customerByIdentityProviderId',
        mockGraphqlClient,
      );
    });

    describe('with resolve info', () => {
      it('should match snapshot', async () => {
        await service.customerByIdentityProviderId(
          mockCustomerResolveInfo,
          identityProviderId,
        );
      });
    });
    describe('without resolve info', () => {
      it('should match snapshot', async () => {
        await service.customerByIdentityProviderId(null, identityProviderId);
      });
    });
  });

  describe('customerPaymentInstrumentCreate', () => {
    const input = {
      alias: 'testAlias',
      nonce: 'testNonce',
    } as CustomerPaymentInstrumentCreateDto;

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerPaymentInstrumentCreate',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerPaymentInstrumentCreate(
        mockPaymentInstrumentResolveInfo,
        testCustomerUuid,
        input,
      );
    });
  });

  describe('customerServiceConfig', () => {
    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerServiceConfig', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerServiceConfig();
    });
  });

  describe('customerDelete', () => {
    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerDelete', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerDelete(testCustomerUuid);
    });
  });

  describe('currentCustomerByUsername', () => {
    const username = 'testUsername';

    beforeEach(() => {
      mockGraphqlClientExecuteQuery('customerByUsername', mockGraphqlClient);
    });

    describe('with resolve info', () => {
      it('should match snapshot', async () => {
        await service.currentCustomerByUsername(
          mockCustomerResolveInfo,
          username,
        );
      });
    });
    describe('without resolve info', () => {
      it('should match snapshot', async () => {
        await service.currentCustomerByUsername(null, username);
      });
    });
  });

  describe('customerUpdate', () => {
    const input = {
      email: 'tetsuo000+2222222222420@gmail.com',
      demographics: {
        birthday: '1977-12-14',
      },
    } as Partial<ICustomerInternal>;

    beforeEach(() => {
      mockGraphqlClientExecuteQuery('customerUpdate', mockGraphqlClient);
    });
    describe('with resolve info', () => {
      it('should match snapshot', async () => {
        await service.customerUpdate(
          mockCustomerResolveInfo,
          testCustomerUuid,
          input,
        );
      });
    });
    describe('without resolve info', () => {
      it('should match snapshot', async () => {
        await service.customerUpdate(null, testCustomerUuid, input);
      });
    });
  });

  describe('customerEmailVerificationSend', () => {
    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerEmailVerificationSend',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerEmailVerificationSend(testCustomerUuid);
    });
  });

  describe('customerPaymentInstrumentUpdate', () => {
    const input = {
      alias: 'testAlias',
      nonce: 'testNonce',
    } as CustomerPaymentInstrumentUpdateDto;

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerPaymentInstrumentUpdate',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerPaymentInstrumentUpdate(
        mockPaymentInstrumentResolveInfo,
        input,
      );
    });
  });

  describe('customerPaymentInstrumentDelete', () => {
    const uuid = 'testUuid';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerPaymentInstrumentDelete',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerPaymentInstrumentDelete(
        mockPaymentInstrumentResolveInfo,
        testCustomerUuid,
        uuid,
      );
    });
  });

  describe('customerCreate', () => {
    const input = {
      email: 'tetsuo000+2222222222420@gmail.com',
      demographics: {
        birthday: '1977-12-14',
      },
    } as Partial<ICustomerInternal>;

    beforeEach(() => {
      mockGraphqlClientExecuteQuery('customerCreate', mockGraphqlClient);
    });
    describe('with resolve info', () => {
      it('should match snapshot', async () => {
        await service.customerCreate(mockCustomerResolveInfo, input);
      });
    });
    describe('without resolve info', () => {
      it('should match snapshot', async () => {
        await service.customerCreate(null, input);
      });
    });
  });

  describe('customerRegistrationComplete', () => {
    const input: CustomerRegistrationCompleteDto = {
      acceptAgreement: [
        'testTermsAndConditionsAgreementUuid',
        'testPrivacyAgreementUuid',
      ],
      email: 'ryan.oboril@stuzo.com',
      name: {
        first: 'Ryan',
        middle: 'H',
        last: 'Test',
        title: 'Sir',
        suffix: 'TheFirst',
      },
      demographics: {
        birthday: '1977-12-14',
        gender: 'm',
        zipCode: '12345',
        customDemographics: {},
      },
      defaultLocationId: 5,
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerRegistrationComplete',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerRegistrationComplete(
        mockCustomerResolveInfo,
        testCustomerUuid,
        input,
      );
    });
  });

  describe('customerPinCodeSet', () => {
    const input = {
      pinCode: {
        pinCode: 'testPinCode',
      },
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerPinCodeSet', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerPinCodeSet(testCustomerUuid, input);
    });
  });

  describe('customerPinCodeReset', () => {
    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerPinCodeReset', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerPinCodeReset(testCustomerUuid);
    });
  });

  describe('customerPinCodeChange', () => {
    const input = {
      oldPinCode: {
        pinCode: '1112',
      },
      newPinCode: {
        pinCode: '4321',
      },
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerPinCodeChange', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerPinCodeChange(testCustomerUuid, input);
    });
  });

  describe('customerPinCodeVerify', () => {
    const input = {
      pinCode: {
        pinCode: 'testPinCode',
      },
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerPinCodeVerify', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerPinCodeVerify(testCustomerUuid, input);
    });
  });
});
