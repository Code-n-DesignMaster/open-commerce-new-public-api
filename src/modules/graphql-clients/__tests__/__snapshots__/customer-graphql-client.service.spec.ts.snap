// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerGraphqlClientService clientPaymentToken should match snapshot: for operation 1`] = `
"query clientPaymentToken($customerID: ID) {
 clientPaymentToken(customerID: $customerID) {
  token
 }
}"
`;

exports[`CustomerGraphqlClientService clientPaymentToken should match snapshot: for parameters 1`] = `
Object {
  "customerID": "testCustomerId",
}
`;

exports[`CustomerGraphqlClientService currentCustomerByUsername with resolve info should match snapshot: for operation 1`] = `
"query customerByUsername($username: String!) {
 customerByUsername(username: $username) {
  pageInfo {
   hasNextPage
   hasPreviousPage
  }
  edges {
   cursor
   node {
    checkNumber
    uuid
    startedAt
    lastUpdatedAt
    completedAt
    additionalReceiptData
    amount {
     displayPrice
     price
    }
    items {
     itemType
     itemDescription
     qty
     amount {
      price
      displayPrice
     }
    }
    paymentProviderTransactionId
    tax {
     price
     displayPrice
    }
    paymentInfo {
     cardPANPrint
     cardType
     hostAuthNumber
    }
    powercard {
     uuid
     isPhysical
     cardNumber
     status
     cardType
     cardAlias
     imagePack {
      uuid
      fullsizeImages {
       url
       width
       height
      }
      thumbnailImages {
       url
       width
       height
      }
      name
     }
     gameChips
     videoChips
     rewardChips
     attractionChips
     tickets
     rewardPoints
     pointsToNextReward
     isRegisteredReward
     easyRechargeEnabled
     walletPass {
      pkPassUrl
      googlePayUrl
     }
    }
    purchaseType
   }
  }
  id
  uuid
 }
}"
`;

exports[`CustomerGraphqlClientService currentCustomerByUsername with resolve info should match snapshot: for parameters 1`] = `
Object {
  "username": "testUsername",
}
`;

exports[`CustomerGraphqlClientService currentCustomerByUsername without resolve info should match snapshot: for operation 1`] = `
"query customerByUsername($username: String!) {
 customerByUsername(username: $username) {

  id
  uuid
  isNewCustomer
  email
  status
  username

 }
}"
`;

exports[`CustomerGraphqlClientService currentCustomerByUsername without resolve info should match snapshot: for parameters 1`] = `
Object {
  "username": "testUsername",
}
`;

exports[`CustomerGraphqlClientService customerByIdentityProviderId with resolve info should match snapshot: for operation 1`] = `
"query customerByIdentityProviderId($identityProviderId: ID!) {
 customerByIdentityProviderId(identityProviderId: $identityProviderId) {
  pageInfo {
   hasNextPage
   hasPreviousPage
  }
  edges {
   cursor
   node {
    checkNumber
    uuid
    startedAt
    lastUpdatedAt
    completedAt
    additionalReceiptData
    amount {
     displayPrice
     price
    }
    items {
     itemType
     itemDescription
     qty
     amount {
      price
      displayPrice
     }
    }
    paymentProviderTransactionId
    tax {
     price
     displayPrice
    }
    paymentInfo {
     cardPANPrint
     cardType
     hostAuthNumber
    }
    powercard {
     uuid
     isPhysical
     cardNumber
     status
     cardType
     cardAlias
     imagePack {
      uuid
      fullsizeImages {
       url
       width
       height
      }
      thumbnailImages {
       url
       width
       height
      }
      name
     }
     gameChips
     videoChips
     rewardChips
     attractionChips
     tickets
     rewardPoints
     pointsToNextReward
     isRegisteredReward
     easyRechargeEnabled
     walletPass {
      pkPassUrl
      googlePayUrl
     }
    }
    purchaseType
   }
  }
  id
  uuid
 }
}"
`;

exports[`CustomerGraphqlClientService customerByIdentityProviderId with resolve info should match snapshot: for parameters 1`] = `
Object {
  "identityProviderId": "testIdentityProviderId",
}
`;

exports[`CustomerGraphqlClientService customerByIdentityProviderId without resolve info should match snapshot: for operation 1`] = `
"query customerByIdentityProviderId($identityProviderId: ID!) {
 customerByIdentityProviderId(identityProviderId: $identityProviderId) {

  id
  uuid
  isNewCustomer
  email
  status
  username

 }
}"
`;

exports[`CustomerGraphqlClientService customerByIdentityProviderId without resolve info should match snapshot: for parameters 1`] = `
Object {
  "identityProviderId": "testIdentityProviderId",
}
`;

exports[`CustomerGraphqlClientService customerCreate with resolve info should match snapshot: for operation 1`] = `
"mutation customerCreate($input: CustomerCreate!) {
 customerCreate(input: $input) {
  pageInfo {
   hasNextPage
   hasPreviousPage
  }
  edges {
   cursor
   node {
    checkNumber
    uuid
    startedAt
    lastUpdatedAt
    completedAt
    additionalReceiptData
    amount {
     displayPrice
     price
    }
    items {
     itemType
     itemDescription
     qty
     amount {
      price
      displayPrice
     }
    }
    paymentProviderTransactionId
    tax {
     price
     displayPrice
    }
    paymentInfo {
     cardPANPrint
     cardType
     hostAuthNumber
    }
    powercard {
     uuid
     isPhysical
     cardNumber
     status
     cardType
     cardAlias
     imagePack {
      uuid
      fullsizeImages {
       url
       width
       height
      }
      thumbnailImages {
       url
       width
       height
      }
      name
     }
     gameChips
     videoChips
     rewardChips
     attractionChips
     tickets
     rewardPoints
     pointsToNextReward
     isRegisteredReward
     easyRechargeEnabled
     walletPass {
      pkPassUrl
      googlePayUrl
     }
    }
    purchaseType
   }
  }
  id
  uuid
 }
}"
`;

exports[`CustomerGraphqlClientService customerCreate with resolve info should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "demographics": Object {
      "birthday": "1977-12-14",
    },
    "email": "tetsuo000+2222222222420@gmail.com",
  },
}
`;

exports[`CustomerGraphqlClientService customerCreate without resolve info should match snapshot: for operation 1`] = `
"mutation customerCreate($input: CustomerCreate!) {
 customerCreate(input: $input) {

  id
  uuid
  isNewCustomer
  email
  status
  username

 }
}"
`;

exports[`CustomerGraphqlClientService customerCreate without resolve info should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "demographics": Object {
      "birthday": "1977-12-14",
    },
    "email": "tetsuo000+2222222222420@gmail.com",
  },
}
`;

exports[`CustomerGraphqlClientService customerDelete should match snapshot: for operation 1`] = `
"mutation customerDelete($customerUuid: ID!) {
 customerDelete(customerUuid: $customerUuid)
}"
`;

exports[`CustomerGraphqlClientService customerDelete should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
}
`;

exports[`CustomerGraphqlClientService customerEmailVerificationSend should match snapshot: for operation 1`] = `
"mutation customerEmailVerificationSend($customerUuid: ID!) {
 customerEmailVerificationSend(customerUuid: $customerUuid)
}"
`;

exports[`CustomerGraphqlClientService customerEmailVerificationSend should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
}
`;

exports[`CustomerGraphqlClientService customerPaymentInstrumentCreate should match snapshot: for operation 1`] = `
"mutation customerPaymentInstrumentCreate($customerUuid: ID!, $input: CustomerPaymentInstrumentCreate!) {
 customerPaymentInstrumentCreate(customerUuid: $customerUuid, input: $input) {
  uuid
  alias
  providerNickName
  accountNumber
  address {
   alias
   street1
   street2
   city
   state
   zipCode
   geoLocation {
    latitude
    longitude
   }
  }
  paymentType
  expiration {
   expirationYear
   expirationMonth
  }
  status
  isDefault
  displayNumber
 }
}"
`;

exports[`CustomerGraphqlClientService customerPaymentInstrumentCreate should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "alias": "testAlias",
    "nonce": "testNonce",
  },
}
`;

exports[`CustomerGraphqlClientService customerPaymentInstrumentDelete should match snapshot: for operation 1`] = `
"mutation customerPaymentInstrumentDelete($customerUuid: ID!, $uuid: ID!) {
 customerPaymentInstrumentDelete(customerUuid: $customerUuid, uuid: $uuid) {
  uuid
  alias
  providerNickName
  accountNumber
  address {
   alias
   street1
   street2
   city
   state
   zipCode
   geoLocation {
    latitude
    longitude
   }
  }
  paymentType
  expiration {
   expirationYear
   expirationMonth
  }
  status
  isDefault
  displayNumber
 }
}"
`;

exports[`CustomerGraphqlClientService customerPaymentInstrumentDelete should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "uuid": "testUuid",
}
`;

exports[`CustomerGraphqlClientService customerPaymentInstrumentUpdate should match snapshot: for operation 1`] = `
"mutation customerPaymentInstrumentUpdate($input: CustomerPaymentInstrumentUpdate!) {
 customerPaymentInstrumentUpdate(input: $input) {
  uuid
  alias
  providerNickName
  accountNumber
  address {
   alias
   street1
   street2
   city
   state
   zipCode
   geoLocation {
    latitude
    longitude
   }
  }
  paymentType
  expiration {
   expirationYear
   expirationMonth
  }
  status
  isDefault
  displayNumber
 }
}"
`;

exports[`CustomerGraphqlClientService customerPaymentInstrumentUpdate should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "alias": "testAlias",
    "nonce": "testNonce",
  },
}
`;

exports[`CustomerGraphqlClientService customerPinCodeChange should match snapshot: for operation 1`] = `
"mutation customerPinCodeChange($customerUuid: ID!, $input: CustomerPinCodeChange!) {
 customerPinCodeChange(customerUuid: $customerUuid, input: $input)
}"
`;

exports[`CustomerGraphqlClientService customerPinCodeChange should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "newPinCode": Object {
      "pinCode": "4321",
    },
    "oldPinCode": Object {
      "pinCode": "1112",
    },
  },
}
`;

exports[`CustomerGraphqlClientService customerPinCodeReset should match snapshot: for operation 1`] = `
"mutation customerPinCodeReset($customerUuid: ID!) {
 customerPinCodeReset(customerUuid: $customerUuid)
}"
`;

exports[`CustomerGraphqlClientService customerPinCodeReset should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
}
`;

exports[`CustomerGraphqlClientService customerPinCodeSet should match snapshot: for operation 1`] = `
"mutation customerPinCodeSet($customerUuid: ID!, $input: CustomerPinCodeSet!) {
 customerPinCodeSet(customerUuid: $customerUuid, input: $input)
}"
`;

exports[`CustomerGraphqlClientService customerPinCodeSet should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "pinCode": Object {
      "pinCode": "testPinCode",
    },
  },
}
`;

exports[`CustomerGraphqlClientService customerPinCodeVerify should match snapshot: for operation 1`] = `
"mutation customerPinCodeVerify($customerUuid: ID!, $input: CustomerPinCodeVerify!) {
 customerPinCodeVerify(customerUuid: $customerUuid, input: $input)
}"
`;

exports[`CustomerGraphqlClientService customerPinCodeVerify should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "pinCode": Object {
      "pinCode": "testPinCode",
    },
  },
}
`;

exports[`CustomerGraphqlClientService customerRegistrationComplete should match snapshot: for operation 1`] = `
"mutation customerRegistrationComplete($customerUuid: ID!, $input: CustomerRegistrationComplete!) {
 customerRegistrationComplete(customerUuid: $customerUuid, input: $input) {
  pageInfo {
   hasNextPage
   hasPreviousPage
  }
  edges {
   cursor
   node {
    checkNumber
    uuid
    startedAt
    lastUpdatedAt
    completedAt
    additionalReceiptData
    amount {
     displayPrice
     price
    }
    items {
     itemType
     itemDescription
     qty
     amount {
      price
      displayPrice
     }
    }
    paymentProviderTransactionId
    tax {
     price
     displayPrice
    }
    paymentInfo {
     cardPANPrint
     cardType
     hostAuthNumber
    }
    powercard {
     uuid
     isPhysical
     cardNumber
     status
     cardType
     cardAlias
     imagePack {
      uuid
      fullsizeImages {
       url
       width
       height
      }
      thumbnailImages {
       url
       width
       height
      }
      name
     }
     gameChips
     videoChips
     rewardChips
     attractionChips
     tickets
     rewardPoints
     pointsToNextReward
     isRegisteredReward
     easyRechargeEnabled
     walletPass {
      pkPassUrl
      googlePayUrl
     }
    }
    purchaseType
   }
  }
  id
  uuid
 }
}"
`;

exports[`CustomerGraphqlClientService customerRegistrationComplete should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "acceptAgreement": Array [
      "testTermsAndConditionsAgreementUuid",
      "testPrivacyAgreementUuid",
    ],
    "defaultLocationId": 5,
    "demographics": Object {
      "birthday": "1977-12-14",
      "customDemographics": Object {},
      "gender": "m",
      "zipCode": "12345",
    },
    "email": "ryan.oboril@stuzo.com",
    "name": Object {
      "first": "Ryan",
      "last": "Test",
      "middle": "H",
      "suffix": "TheFirst",
      "title": "Sir",
    },
  },
}
`;

exports[`CustomerGraphqlClientService customerServiceConfig should match snapshot: for operation 1`] = `
"query customerServiceConfig {
 customerServiceConfig {
  maxAllowedPaymentCards
 }
}"
`;

exports[`CustomerGraphqlClientService customerServiceConfig should match snapshot: for parameters 1`] = `undefined`;

exports[`CustomerGraphqlClientService customerUpdate with resolve info should match snapshot: for operation 1`] = `
"mutation customerUpdate($customerUuid: ID!, $input: CustomerUpdate!) {
 customerUpdate(customerUuid: $customerUuid, input: $input) {
  pageInfo {
   hasNextPage
   hasPreviousPage
  }
  edges {
   cursor
   node {
    checkNumber
    uuid
    startedAt
    lastUpdatedAt
    completedAt
    additionalReceiptData
    amount {
     displayPrice
     price
    }
    items {
     itemType
     itemDescription
     qty
     amount {
      price
      displayPrice
     }
    }
    paymentProviderTransactionId
    tax {
     price
     displayPrice
    }
    paymentInfo {
     cardPANPrint
     cardType
     hostAuthNumber
    }
    powercard {
     uuid
     isPhysical
     cardNumber
     status
     cardType
     cardAlias
     imagePack {
      uuid
      fullsizeImages {
       url
       width
       height
      }
      thumbnailImages {
       url
       width
       height
      }
      name
     }
     gameChips
     videoChips
     rewardChips
     attractionChips
     tickets
     rewardPoints
     pointsToNextReward
     isRegisteredReward
     easyRechargeEnabled
     walletPass {
      pkPassUrl
      googlePayUrl
     }
    }
    purchaseType
   }
  }
  id
  uuid
 }
}"
`;

exports[`CustomerGraphqlClientService customerUpdate with resolve info should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "demographics": Object {
      "birthday": "1977-12-14",
    },
    "email": "tetsuo000+2222222222420@gmail.com",
  },
}
`;

exports[`CustomerGraphqlClientService customerUpdate without resolve info should match snapshot: for operation 1`] = `
"mutation customerUpdate($customerUuid: ID!, $input: CustomerUpdate!) {
 customerUpdate(customerUuid: $customerUuid, input: $input) {

  id
  uuid
  isNewCustomer
  email
  status
  username

 }
}"
`;

exports[`CustomerGraphqlClientService customerUpdate without resolve info should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "input": Object {
    "demographics": Object {
      "birthday": "1977-12-14",
    },
    "email": "tetsuo000+2222222222420@gmail.com",
  },
}
`;
